### DISCUSSION TITLE
[@types/rtlcss] doesn't represent 2nd-4th parameters to rtlcss() function

### DISCUSSION THREAD

**Discussion by catrope** [DefinitelyTyped/DefinitelyTyped#59398] [Issues with a @types package]
`rtlcss` exposes an object that has a `.process()` and a `.configure()` method, but is also a function that returns a PostCSS plugin. This function takes three arguments (`options`, `plugins` and `hooks`) as documented [[https://rtlcss.com/learn/usage-guide/install/|here]] and visible [[https://github.com/MohammadYounes/rtlcss/blob/1b3409f7477a8d0216c740a21296e3bee591fe0c/lib/rtlcss.js#L15|here]]. It also has a static property `postcss: true`, to mark it as a PostCSS plugin.

However, the type definitions for `rtlcss` in this package use the `PluginCreator` generic type from `postcss`, which includes the function and the `postcss: true` property, but assumes that functions will only have one parameter. As a consequence, the type definitions indicate that the function only takes the `options` parameter, and there is no typing for the `plugins` and `hooks` parameters.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
see #59439
